main.go

```diff
@@ -76,7 +76,14 @@ func registerSIGINTHandler(mountPoint string) {
 }
 
 func getUserAgent(appName string) string {
-	return strings.TrimSpace(fmt.Sprintf("gcsfuse/%s %s %s", getVersion(), appName, os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE")))
+	if len(os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE")) > 0 {
+		userAgent := fmt.Sprintf("gcsfuse/%s %s (GPN:gcsfuse-%s)", getVersion(), appName, os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE"))
+		return strings.Join(strings.Fields(userAgent), " ")
```

Comment:
why do we need strings.Join here ?

Reply:
Line 81 is to remove any redundant spaces from the string incase appname is not present. strings.Fields will split the string around spaces and strings.Join will join the fields into a string.